From 42e25f7e69de327ed74b423a337d062e9a5b0b80 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Horia=20Geant=C4=83?= <horia.geanta@nxp.com>
Date: Wed, 20 Mar 2019 08:30:25 +0200
Subject: [PATCH 115/235] crypto: caam - fix building job ring as module
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

In case job ring is built as module, the shared descriptors libraries
also have to be built as modules - otherwise the linking will fail:
[...]
ERROR: "cnstr_shdsc_rfc4106_encap" [drivers/crypto/caam/caam_jr.ko] undefined!
ERROR: "cnstr_shdsc_aead_null_decap" [drivers/crypto/caam/caam_jr.ko] undefined!
ERROR: "cnstr_shdsc_gcm_encap" [drivers/crypto/caam/caam_jr.ko] undefined!
[...]

Fixes: 0f487805d34d ("crypto: caam - fix duplicate files linking")
Signed-off-by: Horia GeantÄƒ <horia.geanta@nxp.com>
[Xulin: Original patch taken from NXP LSDK-19.06.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/crypto/caam/Kconfig | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/crypto/caam/Kconfig b/drivers/crypto/caam/Kconfig
index 321217fc8150..5c2005ca3623 100644
--- a/drivers/crypto/caam/Kconfig
+++ b/drivers/crypto/caam/Kconfig
@@ -2,10 +2,10 @@ config CRYPTO_DEV_FSL_CAAM_COMMON
 	tristate
 
 config CRYPTO_DEV_FSL_CAAM_CRYPTO_API_DESC
-	bool
+	tristate
 
 config CRYPTO_DEV_FSL_CAAM_AHASH_API_DESC
-	bool
+	tristate
 
 config CRYPTO_DEV_FSL_CAAM
 	tristate "Freescale CAAM-Multicore platform driver backend"
-- 
2.17.1

